src/coreclr/hosts/coreshim/CoreShim.cpp

```diff
@@ -106,8 +105,82 @@ namespace Utility
 
         return S_OK;
     }
+
+    HRESULT GetCoreShimDirectory(_Inout_ std::wstring &dir)
+    {
+        HMODULE hModule;
+        BOOL res = ::GetModuleHandleExW(
+            GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+            reinterpret_cast<LPCWSTR>(&TryGetEnvVar),
+            &hModule);
+        if (res == FALSE)
+            return HRESULT_FROM_WIN32(::GetLastError());
+
+        std::wstring path;
+        size_t dwModuleFileName = MAX_PATH / 2;
+
+        do
+        {
+            path.resize(dwModuleFileName * 2);
+            dwModuleFileName = GetModuleFileNameW(hModule, (LPWSTR)path.data(), static_cast<DWORD>(path.size()));
+        } while (dwModuleFileName == path.size());
+
+        if (dwModuleFileName == 0)
+            return HRESULT_FROM_WIN32(::GetLastError());
+
+        size_t idx = path.find_last_of(W('\\'));
+        if (idx == std::wstring::npos)
+            return E_UNEXPECTED;
+
+        path.resize(idx + 1);
+        dir = std::move(path);
+        return S_OK;
+    }
+
+    HRESULT GetCoreShimDirectory(_Inout_ std::string &dir)
+    {
+        HRESULT hr;
+
+        std::wstring dir_wide;
+        RETURN_IF_FAILED(GetCoreShimDirectory(dir_wide));
+
+        dir = ConvertWideToUtf8(dir_wide);
+
+        return S_OK;
+    }
 }
 
+//
+// BEGIN hostpolicy mock
```

Comment:
Boo. This is wrong. I will update this to use the new hostpolicy mock class that I moved to a common location.